src/network/message.rs

```diff
@@ -34,17 +35,40 @@ use consensus::encode::MAX_VEC_SIZE;
 
 /// Serializer for command string
 #[derive(PartialEq, Eq, Clone, Debug)]
-pub struct CommandString(pub String);
+pub struct CommandString(Cow<'static, str>);
+
+impl fmt::Display for CommandString {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.write_str(self.0.as_ref())
+    }
+}
+
+impl From<&'static str> for CommandString {
+    fn from(f: &'static str) -> Self {
+        CommandString(f.into())
+    }
+}
+
+impl From<String> for CommandString {
+    fn from(f: String) -> Self {
+        CommandString(f.into())
+    }
+}
+
+impl AsRef<str> for CommandString {
+    fn as_ref(&self) -> &str {
+        self.0.as_ref()
+    }
+}
 
 impl Encodable for CommandString {
     #[inline]
     fn consensus_encode<S: io::Write>(
         &self,
         s: S,
     ) -> Result<usize, encode::Error> {
-        let &CommandString(ref inner_str) = self;
         let mut rawbytes = [0u8; 12];
-        let strbytes = inner_str.as_bytes();
+        let strbytes = self.0.as_bytes();
         if strbytes.len() > 12 {
             panic!("Command string longer than 12 bytes");
```

Comment:
Would it not be an easy attack vector to let the node panic?

Reply:
That's a good point. TryFrom is Rust 1.34. Hmm yeah then I propose we just truncate them. We already have this code that uses the 12-byte array. I'll add a commit.

Reply:
I added another commit with a test serializing an oversized command string. I think is best like this. Lateron when we have a MSRV of > 1.34, we can upgrade to TryFrom.